src/NovaDependencyContainer.php

```diff
@@ -153,6 +153,23 @@ public function resolveForDisplay($resource, $attribute = null)
                 continue;
             }
 
+            if (is_object($resource->{$dependency['property']}) && is_array($dependency['value'])) {
+                if (array_key_exists('value', $dependency) && in_array($resource->{$dependency['property']}->id, $dependency['value'])) {
+                    $this->meta['dependencies'][$index]['satisfied'] = true;
+                    continue;
+                }
+            }
+
+            if (is_iterable($resource->{$dependency['property']})) {
+                $filtered = $resource->{$dependency['property']}->filter(function ($value, $key) use (&$dependency) {
+                    return $value->id == $dependency['value'];
+                });
+                if (array_key_exists('value', $dependency) && $filtered->isNotEmpty()) {
+                    $this->meta['dependencies'][$index]['satisfied'] = true;
+                    continue;
+                }
+            }
+
             if (array_key_exists('value', $dependency)) {
```

Comment:
Same sort of comment as above, let's just expand the current dependsOn logic to be able to support arrays instead of duplicating it almost completely.